### DISCUSSION TITLE
Prisma 2 Lambda Size 55mb...??

### DISCUSSION THREAD

**Discussion by hmmhmmhm** [prisma/prisma#8067] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/38493722/124388043-e160b900-dd1b-11eb-8d55-a79a38e00638.png)

![image](https://user-images.githubusercontent.com/38493722/124388000-a5c5ef00-dd1b-11eb-80cf-16b25f636ae6.png)

![image](https://user-images.githubusercontent.com/38493722/124388163-95624400-dd1c-11eb-9893-8e6253834063.png)

The size of prisma uploaded to lambda is too big, is this normal?
`@prima/client` is using prisma again imputedly, so prisma seems to be included in the build output as well.

Is my bundle size normal? Is there any way to reduce it further?

**Comment by janpio** [comment, marked as answer]
Yes, you can absolutely configure this to be less. Can you check what is actually included in the generated archive?

Here is the configuration of our end to end test project where we monitor the size: https://github.com/prisma/e2e-tests/blob/e80772020285232745280f7e0b15e3ccf7fe88f8/platforms-serverless/serverless-framework-lambda/serverless.yml#L19-L25

**Reply by hmmhmmhm** [reply]
![image](https://user-images.githubusercontent.com/38493722/124388439-ed4d7a80-dd1d-11eb-9d17-64a77a28bf0c.png)
I applied the patterns but the bundle size is the same, which one should I look at? Do you want me to share my build copy?

**Reply by hmmhmmhm** [reply]
![image](https://user-images.githubusercontent.com/38493722/124388653-c04d9780-dd1e-11eb-913a-1d0296395e27.png)
@janpio 
I was able to check the query-engine-darwin file everywhere. Do I need to remove any files that are different from the .prisma path?

**Reply by janpio** [reply]
The whole `@prisma/engines` folder should not be present, same as `prisma` as those are both part of the CLI- which is a devDependency and should not be included in the archive. Only a copy of the query engine inside the generated `.prisma` folder should be present.

What does your `schema.prisma` look like?
What Node version are you using?

**Reply by hmmhmmhm** [reply]
schema.prisma
<img width="481" alt="스크린샷 2021-07-04 오후 11 29 36" src="https://user-images.githubusercontent.com/38493722/124388877-baa48180-dd1f-11eb-95d9-1ceed5bfd482.png">

node version
<img width="563" alt="스크린샷 2021-07-04 오후 11 30 11" src="https://user-images.githubusercontent.com/38493722/124388886-c8f29d80-dd1f-11eb-823f-7ef995927b40.png">

serverless.yml
<img width="600" alt="스크린샷 2021-07-04 오후 11 31 52" src="https://user-images.githubusercontent.com/38493722/124388946-0e16cf80-dd20-11eb-881a-e2b7d5fa1c5a.png">

**Reply by hmmhmmhm** [reply]
<img width="691" alt="스크린샷 2021-07-04 오후 11 41 48" src="https://user-images.githubusercontent.com/38493722/124389293-6ef2d780-dd21-11eb-971d-8f8a126e6576.png">
I added more patterns and the bundle size has been reduced to 5MB, is it okay to do this?

**Reply by janpio** [reply]
5MB is to small, at least one query Engine needs to be included for the Client to work. That usually never goes <15MB, so I would say that is not the case. But just look at the zip file - the `.prisma/client` folder needs to include one Query Engine.

I think your problems are caused by https://github.com/dougmoscrop/serverless-plugin-include-dependencies. It disabled the default logic of Serverless Framework, which is what the `patterns` are for. Probably that is a side effect of using this plugin.

**Reply by hmmhmmhm** [reply]
![image](https://user-images.githubusercontent.com/38493722/124390005-a3b45e00-dd24-11eb-99db-9fbfc8ad5ed8.png)

After deleting the module, the rest of the files started to go up. However, in my project file (project file before it was bundled) node_modules does not have Query Engine in the .prima/client folder. What should I do?

**Reply by hmmhmmhm** [reply]
oh sorry it's my bad, it actually exist it, but i'm trying to see other projects... 😭 anyway my query engine builded to darwin. so i changed package pattern, and now i'll trying to check is this working
<img width="448" alt="스크린샷 2021-07-05 오전 12 25 23" src="https://user-images.githubusercontent.com/38493722/124390636-7f0db580-dd27-11eb-9c03-0bddbe254f91.png">

**Reply by hmmhmmhm** [reply]
<img width="485" alt="스크린샷 2021-07-05 오전 12 32 55" src="https://user-images.githubusercontent.com/38493722/124390908-baf54a80-dd28-11eb-9f19-9e8d2690e1e2.png">

nahh.. i founded my schema.prisma need to add `rhel-openssl-1.0.x`
i restored serverless.yml files, and i 'll trying again...

<img width="624" alt="스크린샷 2021-07-05 오전 12 32 47" src="https://user-images.githubusercontent.com/38493722/124390916-cf394780-dd28-11eb-9183-9b4dd09eac26.png">

**Reply by hmmhmmhm** [reply]
Solved it...! so much thanks @janpio ...! you saved me..!!

**Reply by janpio** [reply]
Ha, that makes me happy. Can you share your final configuration and setup?

(I should have mentioned the `binaryTargets` before - it is required for Lamba and alike where you build on one platform, then upload to another. At least our error message was helpful.)

**Reply by hmmhmmhm** [reply]
@janpio 
I shared my final results on GitHub!
Thank you again for your help 😊
https://github.com/hmmhmmhm/serverless-prisma

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
